### PR TITLE
add mode for Solidity language

### PR DESCRIPTION
**Pull request by tom2drum**
*Issue #, if available:*
#3697 and #3413 

*Description of changes:*
Implemented mode for highlighting [Solidity](https://docs.soliditylang.org) language. Mode is completely based on the [ace-mode-solidity](https://github.com/raphaelhuefner/ace-mode-solidity) package

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/mode/solidity_highlight_rules.js [hunk 1]
```diff
@@ -0,0 +1,358 @@
+// rules were taken from https://github.com/raphaelhuefner/ace-mode-solidity
+// all credits to @raphaelhuefner
+
+define(function(require, exports, module) {
```
Comment (nightwing) [inline src/mode/solidity_highlight_rules.js:4]: Could you remove the define wrappers, these are not needed anymore.

Comment (tom2drum) [reply src/mode/solidity_highlight_rules.js]: done

**Comment by akoreman** [comment]
The repository these highlighting rules are based on is available under MIT license, so we should include the license block from that package if we want to use that code.

**Comment by tom2drum** [comment]
> The repository these highlighting rules are based on is available under MIT license, so we should include the license block from that package if we want to use that code.

done

**Comment by InspiredGuy** [comment]
hey @tom2drum this looks good, but codecov is complaining about test coverage - could you please add tests for the added language mode? there are some examples, smaller and bigger, in https://github.com/ajaxorg/ace/tree/master/src/mode - e.g. https://github.com/ajaxorg/ace/blob/master/src/mode/coldfusion_test.js or https://github.com/ajaxorg/ace/blob/master/src/mode/javascript_test.js

**Comment by tom2drum** [comment]
Apologies, @InspiredGuy! I've decided to close this PR as I no longer wish to incorporate the library's new mode for Solidity into my project. However, if someone else requires it, they can submit their own pull request along with the required tests and so on.

**Pull request closed without merging**